src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -466,71 +468,58 @@ private int dupeCheckStandardizationStructure(Long dryRunId) throws CmpdRegMolFo
-			newDupeCount = hits.length;
-			for (int hit : hits) {
-				List<StandardizationDryRunCompound> searchResults = StandardizationDryRunCompound
-						.findStandardizationDryRunCompoundsByCdId(hit).getResultList();
-				for (StandardizationDryRunCompound searchResult : searchResults) {
-					if (searchResult.getCorpName().equalsIgnoreCase(dryRunCompound.getCorpName())) {
-						newDupeCount = newDupeCount - 1;
-					} else {
-						if (StringUtils.equals(searchResult.getStereoCategory(),
-								dryRunCompound.getStereoCategory())
-								&& StringUtils.equalsIgnoreCase(searchResult.getStereoComment(),
-										dryRunCompound.getStereoComment())) {
-							if (!firstNewDuplicateHit)
-								newDuplicateCorpNames = newDuplicateCorpNames.concat(";");
-							newDuplicateCorpNames = newDuplicateCorpNames.concat(searchResult.getCorpName());
-							firstNewDuplicateHit = false;
-							logger.info("found new dupe parents - query: '" + dryRunCompound.getCorpName() + "' dupe: '"
-									+ searchResult.getCorpName() + "'");
-							totalNewDuplicateCount++;
-						} else {
-							newDupeCount = newDupeCount - 1;
-							logger.debug("found different stereo codes and comments");
-						}
-					}
-				}
-			}
-			hits = chemStructureService.searchMolStructures(cmpdRegMolecules.get(tmpStructureKey),
-					StructureType.PARENT, SearchType.DUPLICATE_TAUTOMER, -1F, -1);
-			oldDuplicateCount = hits.length;
-			dryRunCompound.setChangedStructure(true);
-			for (int hit : hits) {
-				List<Parent> searchResults = Parent.findParentsByCdId(hit).getResultList();
-				for (Parent searchResult : searchResults) {
-					if (searchResult.getCorpName().equalsIgnoreCase(dryRunCompound.getCorpName())) {
-						oldDuplicateCount = oldDuplicateCount - 1;
-						dryRunCompound.setChangedStructure(false);
-					} else {
-						if (StringUtils.equals(searchResult.getStereoCategory().getName(),
-								dryRunCompound.getStereoCategory())
-								&& StringUtils.equalsIgnoreCase(searchResult.getStereoComment(),
-										dryRunCompound.getStereoComment())) {
-							if (!firstOldDuplicateHit)
-								oldDuplicateCorpNames = oldDuplicateCorpNames.concat(";");
-							oldDuplicateCorpNames = oldDuplicateCorpNames.concat(searchResult.getCorpName());
-							firstOldDuplicateHit = false;
-							logger.info("found old dupe parents - query: '" + dryRunCompound.getCorpName() + "' dupe: "
-									+ searchResult.getCorpName() + "'");
-							// totalExistingDuplicateCount++;
-						} else {
-							oldDuplicateCount = oldDuplicateCount - 1;
-							logger.debug("found different stereo codes and comments");
-						}
-					}
-				}
+			List<StandardizationDryRunCompound> dryRunDupes = StandardizationDryRunCompound.checkForDuplicateStandardizationDryRunCompoundByCdId(dryRunCompound.getId(), hits).getResultList();
+			newDupeCount = dryRunDupes.size();
+			if(newDupeCount > 0) {
+				dryRunCompound.setChangedStructure(true);
```

Comment:
It looks like this line was outdented by one in the previous code, i.e. it was not inside an `if(newDupeCount > ))` block.

It also seems kindof unrelated to the "newDupeCount", and redundant with lines 504-508. Can we remove this line up here, and simply have the checks below on 504-508?

Reply:
Thanks your analysis is correct. I made some code moves and accidentally left this here.